File: androidApp/src/main/kotlin/com/chaoticware/discrobble/android/auth/AuthProvider.kt
package com.chaoticware.discrobble.android.auth

enum class AuthProvider(
    val rawValue: String,
    val displayName: String,
) {
    LASTFM(
        rawValue = "lastfm",
        displayName = "Last.fm",
    ),
    DISCOGS(
        rawValue = "discogs",
        displayName = "Discogs",
    );

    companion object {
        fun fromRawValue(rawValue: String): AuthProvider? {
            return entries.firstOrNull { provider ->
                provider.rawValue == rawValue
            }
        }
    }
}

File: iosApp/Discrobble/Auth/AuthProvider.swift
import Foundation

enum AuthProvider: String, CaseIterable, Codable, Identifiable {
    case lastfm
    case discogs

    var id: String {
        rawValue
    }

    var displayName: String {
        switch self {
        case .lastfm:
            return "Last.fm"
        case .discogs:
            return "Discogs"
        }
    }
}
